### ISSUE TITLE
Can't get the same result

### ISSUE THREAD

**Issue by CodingPerson** [yueyu1030/COSINE#9]
Hello, I ran run.agnews.sh and did not get the same result in the paper. Is it because the parameters are not optimal? If so, how can I adjust it?

thank you

**Comment by CodingPerson** [comment]
![image](https://user-images.githubusercontent.com/45897585/133254462-57edc301-db8f-4c12-ae4d-29dcd412ff4f.png)
First of all, thank you. Secondly, I want to know where the parameters represented by T1 need to be adjusted. I don't quite understand

**Comment by yueyu1030** [comment]
T1 is the number of steps for initialization. Our method is reproducible and has been integrated into this system [https://github.com/JieyuZ2/wrench].
You can also try to tune the learning rate, T3 and \xi. You may need to first tune on the T3 to obtain the best model for initialization, then use the corresponding model for self-training. Thanks!

**Comment by CodingPerson** [comment]
![image](https://user-images.githubusercontent.com/45897585/133361733-00f08933-28d9-4ad4-b510-02ff5a044b05.png)

If I follow the above parameters exactly, can I get the results in the paper? thank you

**Comment by CodingPerson** [comment]
![无标题](https://user-images.githubusercontent.com/45897585/133450090-4ca9ae49-3339-4e74-a322-a8a3d14abbd9.png)
Can you tell us what the parameters in the red circle mean? Especially several "batch" parameters, I can not understand their meaning?
Thank you very much

**Comment by yueyu1030** [comment]
`batch size` is the batch size used for each pass
`real_batch_size` is the batch size used in training (in some cases, your GPU space might be limited, and you can use gradient accumulation steps to simulate large batch. For example, if you set batch_size = 64 and real_batch_size = 256, then the model will be updated after 256/64=4 steps)
`test_batch_size` is the batch size used during the evaluation

**Comment by yueyu1030** [comment]
> ![image](https://user-images.githubusercontent.com/45897585/133361733-00f08933-28d9-4ad4-b510-02ff5a044b05.png)
> 
> If I follow the above parameters exactly, can I get the results in the paper?
> If not, what other parameters should I pay attention to?
> thanks a lot

Here I think you first need to ensure that the init model is good enough. You may need to choose a good ckpt to start self-training. Then, for self-training, the update period and $\xi$ are important parameters to tune.

**Comment by CodingPerson** [comment]
> T1 is the number of steps for initialization. Our method is reproducible and has been integrated into this system [https://github.com/JieyuZ2/wrench] (reproduced by this implementation).
> You can also try to tune the learning rate, T3 and \xi. You may need to first tune on the T3 to obtain the best model for initialization, then use the corresponding model for self-training. Thanks!

**Comment by CodingPerson** [comment]
How can I adjust the parameters to get the results in the paper? For example, if I want to reproduce the results of the AGnews dataset in your paper, since I don't find early Stops in your code, does that mean I need to get the best initialization model through the dev set and then self-train in a similar way later?
Thank you very much for your reply. If it is convenient, could we communicate in Chinese or exchange contact information

**Comment by BD-MF** [comment]
我也有相同的疑惑，论文中的T1、T2、T3具体分别对应代码中的哪些变量呢？
